exchanges/order/limits.go

```diff
@@ -0,0 +1,364 @@
+	// values
+	ErrExchangeLimitNotLoaded = errors.New("exchange Limits not loaded")
+	// ErrPriceExceedsMin is when the price is lower than the minimum price
+	// Limit accepted by the exchange
+	ErrPriceExceedsMin = errors.New("price exceeds minimum Limit")
+	// ErrPriceExceedsMax is when the price is higher than the maximum price
+	// Limit accepted by the exchange
+	ErrPriceExceedsMax = errors.New("price exceeds maximum Limit")
+	// ErrPriceExceedsStep is when the price is not divisible by its step
+	ErrPriceExceedsStep = errors.New("price exceeds step Limit")
+	// ErrAmountExceedsMin is when the amount is lower than the minimum amount
+	// Limit accepted by the exchange
+	ErrAmountExceedsMin = errors.New("amount exceeds minimum Limit")
+	// ErrAmountExceedsMax is when the amount is higher than the maximum amount
+	// Limit accepted by the exchange
+	ErrAmountExceedsMax = errors.New("amount exceeds maximum Limit")
+	// ErrAmountExceedsStep is when the amount is not divisible by its step
+	ErrAmountExceedsStep = errors.New("amount exceeds step Limit")
+	// ErrNotionalValue is when the notional value does not exceed currency pair
+	// requirements
+	ErrNotionalValue = errors.New("total notional value is under minimum Limit")
+	// ErrMarketAmountExceedsMin is when the amount is lower than the minimum
+	// amount Limit accepted by the exchange for a market order
+	ErrMarketAmountExceedsMin = errors.New("market order amount exceeds minimum Limit")
+	// ErrMarketAmountExceedsMax is when the amount is higher than the maximum
+	// amount Limit accepted by the exchange for a market order
+	ErrMarketAmountExceedsMax = errors.New("market order amount exceeds maximum Limit")
+	// ErrMarketAmountExceedsStep is when the amount is not divisible by its
+	// step for a market order
+	ErrMarketAmountExceedsStep = errors.New("market order amount exceeds step Limit")
+
+	errCannotValidateAsset         = errors.New("cannot check Limit asset not loaded")
+	errCannotValidateBaseCurrency  = errors.New("cannot check Limit base currency not loaded")
+	errCannotValidateQuoteCurrency = errors.New("cannot check Limit quote currency not loaded")
+	errExchangeLimitAsset          = errors.New("exchange Limits not found for asset")
+	errExchangeLimitBase           = errors.New("exchange Limits not found for base currency")
+	errExchangeLimitQuote          = errors.New("exchange Limits not found for quote currency")
+	errCannotLoadLimit             = errors.New("cannot load Limit levels not supplied")
+	errInvalidPriceLevels          = errors.New("invalid price levels cannot load Limits")
+	errInvalidAmountLevels         = errors.New("invalid amount levels cannot load Limits")
+)
+
+// ExecutionLimits defines minimum and maximum values in relation to
+// order size, order pricing, total notional values, total maximum orders etc
+// for execution on an exchange.
+type ExecutionLimits struct {
+	m map[asset.Item]map[currency.Code]map[currency.Code]*Limits
+	sync.Mutex
```

Comment:
I think a RWMutex would be better here for now up until a routine is added for updating them every X ticker.  Once these are loaded, we'll mainly be performing reads.